packages/dht/test/integration/ConnectionManager.test.ts

```diff
@@ -0,0 +1,183 @@
+    const appId = 'demo'
+
+    const mockPeerDescriptor1: PeerDescriptor = {
+        peerId: PeerID.fromString("tester1").value,
+        type: NodeType.NODEJS
+    }
+    const mockPeerDescriptor2: PeerDescriptor = {
+        peerId: PeerID.fromString("tester2").value,
+        type: NodeType.NODEJS
+    }
+    const simulator = new Simulator()
+
+    const mockTransport = new SimulatorTransport(mockPeerDescriptor1, simulator)
+    const mockConnectorTransport1 = new SimulatorTransport(mockPeerDescriptor1, simulator)
+    const mockConnectorTransport2 = new SimulatorTransport(mockPeerDescriptor2, simulator)
+
+    beforeAll(async () => {
+
+    })
+
+    it('Can start alone', async () => {
+        const connectionManager = new ConnectionManager({ transportLayer: mockTransport, webSocketHost: 'localhost', webSocketPort: 9991 })
+
+        const result = await connectionManager.start()
+        expect(result.ip).toEqual('localhost')
+        expect(result.openInternet).toEqual(true)
+
+        await connectionManager.stop()
+    })
+
+    it('Throws an async exception if fails to connect to entrypoints', async () => {
+
+        const connectionManager = new ConnectionManager({
+            transportLayer: mockTransport,
+            webSocketPort: 9992, entryPoints: [
+                { peerId: Uint8Array.from([1, 2, 3]), type: NodeType.NODEJS, websocket: { ip: 'localhost', port: 123 } }
+            ]
+        })
+        
+        await expect(connectionManager.start())
+            .rejects
+            .toThrow('Failed to connect to the entrypoints')
+
+        await connectionManager.stop()
+    })
+
+    it('Can probe connectivity in open internet', async () => {
+        const connectionManager1 = new ConnectionManager({ transportLayer: mockTransport, webSocketHost: 'localhost', webSocketPort: 9993 })
+
+        const result = await connectionManager1.start()
+        connectionManager1.enableConnectivity(createPeerDescriptor(result))
+
+        expect(result.ip).toEqual('localhost')
+        expect(result.openInternet).toEqual(true)
+
+        const connectionManager2 = new ConnectionManager({
+            transportLayer: mockConnectorTransport2,
+            webSocketPort: 9994, entryPoints: [
+                { peerId: Uint8Array.from([1, 2, 3]), type: NodeType.NODEJS, websocket: { ip: 'localhost', port: 9993 } }
+            ]
+        })
+       
+        const result2 = await connectionManager2.start()
+        connectionManager2.enableConnectivity(createPeerDescriptor(result2))
+
+        expect(result2.ip).toEqual('127.0.0.1')
+        expect(result2.openInternet).toEqual(true)
+
+        await connectionManager1.stop()
+        await connectionManager2.stop()
+    })
+
+    it('Can send data to other connectionmanager over websocket', async () => {
+        const connectionManager1 = new ConnectionManager({ transportLayer: mockConnectorTransport1, webSocketHost: 'localhost', webSocketPort: 9995 })
```

Comment:
Quite much copy-paste, could extract to helper methods or use `beforeEach`

Reply:
Let's do it

Reply:
Checked the test, and did not find that much copy-paste